### DISCUSSION TITLE
local-storage-state: Overwrites values ​​instead of adding them

### DISCUSSION THREAD

**Discussion by HosseinKrouna** [neuefische/questions#477] [Web] [answered]
Hello everyone, my values ​​are overwritten directly. I would like to add items only. 
I am very grateful for a hint, thank you

This is where the data comes from
```
import useLocalStorageState from "use-local-storage-state";

export default function App({ Component, pageProps }) {
  const [entryData, setEntryData] = useLocalStorageState("entryData", {
    defaultValue: [
      {
        title: "",
        startTime: "",
        endTime: "",
        location: "",
        mapURL: "",
        description: "",
        creator: "",
      },
    ],
  });

  function handleEntryData(data) {
    return setEntryData(data);
  }

  return (
    <>
      <GlobalStyle />
      <Head>
        <title>Capstone Project</title>
      </Head>
      <Component
        {...pageProps}
        entryData={entryData}
        onHandleEntryData={handleEntryData}
      />
    </>
  );
}
```
...and here the data should be included in a list
```

function EventsList({ entryData }) {

  const entryDataToList = [entryData];


  return (
    <>
      <StyledWrapper>
                  ...

        <StyledList>
          {entryDataToList.map(({ title }) => (
            <StyledCardListContainer key={uuid}>
              <li>
                <StyledLink href={"/votepage/"}>{title}</StyledLink>
              </li>
            </StyledCardListContainer>
          ))}
        </StyledList>
        <Link href="/createEventPage/index/">Create Event</Link>
      </StyledWrapper>
    </>
  );
}
```

**Comment by JoergWolff** [comment]
Hi i think you only add the new data to your setter. it should be:
```js
  function handleEntryData(data) {
    return setEntryData(data,...entryData);
  }
```

**Reply by HosseinKrouna** [reply]
Hi, after that I get a `TypeError: entryData is not iterable` when I just add that.

**Reply by JoergWolff** [reply]
Your mapping solution is not correct. Your list needs a key and must use title.uuid for example and not only uuid.
"title" is an object.
Also it seems you have a list in a styled.list.

**Comment by JoergWolff** [comment]
I think you have another problem with your code.
It seems to be another question in another thread.

**Comment by F-Kirchhoff** [comment, marked as answer]
Hey Hossein, your state needs to be an array, and you need to add your new item to the end or beginning of that array, whatever you choose:
```js
function handleEntryData(newDataItem) {
  return setEntryData([...entryData,newDataItem]);
}
```
Here, entryData must be an array (which it is), and you create a new array with the `[]` brackets. Then you spread all entries from your old state into that new array, and you add another entry at the end. By that, your new state is the old one plus another item.
